### PR TITLE
Implement Category Entity and Associated Functionality

### PR DESCRIPTION
**Pull request by MateuszRostek**
Introduction of Category entity with id, name, and description attributes.
Addition of a private Set<Category> field in the Book class for the relationship.
Implementation of CategoryRepository for CRUD operations using JpaRepository.
Development of DTO classes for seamless data transfer.
Expansion of BookMapper and introduction of CategoryMapper for mapping entities to DTOs.
Implementation of CategoryService with CRUD operations in CategoryServiceImpl.
Creation of CategoryController with endpoints for CRUD operations.
Integration of Pagination, Sorting, and Swagger documentation for enhanced API usability.
Implementation of soft delete and using Liquibase for maintaining data integrity.

### CODE REVIEW

File: src/main/java/book/store/service/user/impl/UserServiceImpl.java [hunk 1]
```diff
@@ -24,6 +32,12 @@ public UserResponseDto register(UserRegistrationRequestDto requestDto)
                     "Can't register user - user with this email already exists!");
         }
         User modelUser = userMapper.toModel(requestDto);
+        modelUser.setPassword(passwordEncoder.encode(requestDto.password()));
+        Set<Role> roles = new HashSet<>();
+        Role defaultRole = roleRepository.findByName(DEFAULT_ROLE_NAME)
+                .orElseThrow(() -> new RegistrationException("Can't find default role"));
+        roles.add(defaultRole);
+        modelUser.setRoles(roles);
```
Comment (nowickimj) [inline src/main/java/book/store/service/user/impl/UserServiceImpl.java]: You can simplify such usages with `modelUser.setRoles(Set.of(defaultRole));

**Review by nowickimj — CHANGES_REQUESTED** [review]
I've encountered problems while testing your code - here's the case scenario:
1. I've registered a new user (id = 3):
obraz
2. I've granted user #3 an ADMIN role:
obraz
3. I use API to log in and get the bearer token.
4. GET on /api/categories works but when I try to call POST to create a new category, I got 401.

Please verify if your security configuration is correct.

**Comment by MateuszRostek** [comment]
I can't recreate the problem you have, here is a video, it works fine for me:
https://www.loom.com/share/39024ff4dde74c9697f97bb905d5bc01?sid=f9ce00bb-cbf8-4cdd-a065-ab23a37d9579

**Review by nowickimj — APPROVED** [review]

**Review by MateuszRostek — COMMENTED** [review]
.

**Pull request merged**